src/Kovan-DssSpell.t.sol

```diff
@@ -1278,71 +1698,390 @@ contract DssSpellTest is DSTest, DSMath {
         assertTrue(totalGas <= 8 * MILLION);
     }
 
-    function test_nextCastTime() public {
-        hevm.warp(1606161600); // Nov 23, 20 UTC (could be cast Nov 26)
+    function testSpellIsCast_new_End_authorities() public {
+        assertEq(vat.wards(address(end_old)), 1);
+        assertEq(cat.wards(address(end_old)), 1);
+        assertEq(vow.wards(address(end_old)), 1);
+        assertEq(pot.wards(address(end_old)), 1);
+        assertEq(spotter.wards(address(end_old)), 1);
 
         vote(address(spell));
-        spell.schedule();
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
 
-        uint256 monday_1400_UTC = 1606744800; // Nov 30, 2020
-        uint256 monday_2100_UTC = 1606770000; // Nov 30, 2020
+        // Contracts set
+        assertEq(end.vat(), address(vat));
+        assertEq(end.cat(), address(cat));
+        assertEq(end.dog(), address(dog));
+        assertEq(end.vow(), address(vow));
+        assertEq(end.pot(), address(pot));
+        assertEq(end.spot(), address(spotter));
+
+        // Check end.wait
+        assertEq(end.wait(), end_old.wait());
+
+        assertEq(esmBug.end(), address(end));
+        assertEq(esmAttack.end(), address(end));
+
+        // Check flippers/clippers authorities and osms whitelisting
+        bytes32[] memory ilks = reg.list();
+        for (uint256 i = 0; i < ilks.length; i++) {
+            if (reg.class(ilks[i]) < 3) {
+                FlipAbstract xlip = FlipAbstract(reg.xlip(ilks[i]));
+                assertEq(xlip.wards(address(end)), 1);
+                assertEq(xlip.wards(address(end_old)), 0);
+
+                assertEq(xlip.wards(address(esmAttack)), 1);
+
+                OsmAbstract osm = OsmAbstract(reg.pip(ilks[i]));
+                try osm.bud(address(123)) { // Check is an OSM or Median
```

Comment:
Isn't this actually checking "OSM vs DSValue", not "OSM vs Median"?

Reply:
My comment was meaning that check if the feed is an osm or a median as both have `bud` against the DSValue. It's true in practice we do not use a median without an osm but in theory could happen.